docs/proposals/selinux.md

```diff
@@ -0,0 +1,280 @@
+encompass either of these topics.  There are several problems that the solution for allocation
+depends on:
+
+1.  Users and groups in Kubernetes
+2.  General auth policy in Kubernetes
+3.  [security policy](https://github.com/GoogleCloudPlatform/kubernetes/pull/7893)
+
+### API changes
+
+The [inline container security attributes PR (12823)](https://github.com/kubernetes/kubernetes/pull/12823)
+adds a `pod.Spec.SecurityContext.SELinuxOptions` field.  The change to the API in this proposal is
+the addition of the semantics to this field:
+
+* When the `pod.Spec.SecurityContext.SELinuxOptions` field is set, volumes that support ownership
+management in the Kubelet have their SELinuxContext set from this field.
+
+```go
+package api
+
+type PodSecurityContext struct {
+    // SELinuxOptions captures the SELinux context for all containers in a Pod.  If a container's
+    // SecurityContext.SELinuxOptions field is set, that setting has precedent for that container.
+    //
+    // This field will be used to set the SELinux of volumes that support SELinux label management
+    // by the kubelet.
+    SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
+}
+```
+
+The V1 API is extended with the same semantics:
+
+```go
+package v1
+
+type PodSecurityContext struct {
+    // SELinuxOptions captures the SELinux context for all containers in a Pod.  If a container's
+    // SecurityContext.SELinuxOptions field is set, that setting has precedent for that container.
+    //
+    // This field will be used to set the SELinux of volumes that support SELinux label management
+    // by the kubelet.
+    SELinuxOptions *SELinuxOptions `json:"seLinuxOptions,omitempty"`
+}
+```
+
+#### API backward compatibility
+
+Old pods that do not have the `pod.Spec.SecurityContext.SELinuxOptions` field set will not receive
+SELinux label management for their volumes.  This is acceptable since old clients won't know about
+this field and won't have any expectation of their volumes being managed this way.
+
+The existing backward compatibility semantics for SELinux do not change at all with this proposal.
+
+### Kubelet changes
+
+The Kubelet should be modified to perform SELinux label management when required for a volume.  The
+criteria to activate the kubelet SELinux label management for volumes are:
+
+1.  SELinux integration is enabled in the cluster
+2.  SELinux is enabled on the node
+3.  The `pod.Spec.SecurityContext.SELinuxOptions` field is set
+4.  The volume plugin supports SELinux label management
+
+The `volume.Builder` interface should have a new method added that indicates whether the plugin
+supports SELinux label management:
+
+```go
+package volume
+
+type Builder interface {
+    // other methods omitted
+    SupportsSELinuxLabelManagement() bool
+}
+```
+
+Individual volume plugins are responsible for correctly reporting whether they support label
```

Comment:
Does "support" mean they want the runtime to do :Z or does it mean they handle it themselves and DON'T want auto support?